src/net.cpp

```diff
@@ -455,6 +455,8 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         connect_to.push_back(addrConnect);
     }
 
```

Comment:
The function `AddI2PSessionsIfNeeded` is called without checking if I2P is enabled, potentially leading to unnecessary session creation.